imxrt-iomuxc/src/i2c.rs

```diff
@@ -1,9 +1,9 @@
-//! I2C pad configurations
+//! I2C pad configuration
 
 /// Tag that indicates the SCL signal
-pub struct SCL;
+pub enum SCL {}
```

Comment:
It's one of a few ways to express a zero-sized type. But, this ZST can never (accidentally) be an object, unlike the current approach.

```rust
pub struct SCL; // 1
```

could be instantiated by crate users. Not a big deal, but it's not something they should do.

```rust
pub struct SCL(()); // 2
```

cannot be instantiated by crate users, since they can't see the inner unit. But, us `imxrt-iomuxc` maintainers can still create an instance of it inside our crate, then pass an instance of it through our API. Again, not something we should do, but we're not preventing it.

```rust
pub enum SCL {} // 3
```

cannot be instantiated outside our crate, or inside our crate, since there's no variants. It's the strongest signal of "don't make an instance," since it's intended be a type-level tag.

The only difference I can note is how rustdoc documents these type-level tags. They're either grouped as enums or structs. I've no preference here. If we prefer organizing type tags as `structs`, I'll recommend that we go to option `// 2` to prevent users from creating needless `SCL` objects.

Reply:
This makes sense to me, thanks for explaining. Enums seem fine to me